Companion/main.go

```diff
@@ -31,13 +46,16 @@ func main() {
 	log.Default().SetOutput(logFile)
 
 	// Create exporter
-	promExporter := exporter.NewPrometheusExporter("http://localhost:8080")
-
-	// Create prometheus
-	prom, err := prometheus.NewPrometheusWrapper()
-	if err != nil {
-		fmt.Printf("error preparing prometheus: %s", err)
-		os.Exit(1)
+	promExporter := exporter.NewPrometheusExporter("http://" + frmHostname + ":" + strconv.Itoa(frmPort))
+
+	var prom *prometheus.PrometheusWrapper
+	if runtime.GOOS == "windows" && !noProm {
```

Comment:
I think it would be good to have OS-specific behaviour compiled only for that OS, using Go's conditional compilation (i.e. `foo_linux.go`, `foo_windows.go`). I'll leave a more general comment about Prometheus elsewhere.